django/test/client.py

```diff
@@ -881,6 +889,71 @@ def _parse_json(self, response, **extra):
             )
         return response._json
 
+    def _follow_redirect(
+        self, response, *, data="", content_type="", headers=None, **extra
+    ):
+        """
+        Follow a single redirect contained in response using GET.
+        """
+        assert response.status_code in REDIRECT_STATUS_CODES
```

Comment:
the function expects the response object to be a redirect. Not sure what will happen if it contains something else. I doubt it will produce anything useful. any guidelines to follow regarding assertions ?

Reply:
Sometimes we add assertions to a "tricky" code to avoid regression during refactoring. I don't think that's the case here.